cmd/lib.go

```diff
@@ -36,48 +30,58 @@ var defaultStoreHostTargetPath = "/var/lib/resonance"
 
 func wrapHost(ctx context.Context, hst host.Host) (host.Host, error) {
 	var err error
-	if sudo {
+	switch options {
```

Comment:
I think this logic changes some of the rules, which may have been unintentional.

Before:

- sudo on request
- agent default:
	- localhost: NO (because localhost is faster without an agent)
	- docker: NO (it should be faster with agent, this should be changed to yes)
	- ssh: YES (as it is a billion times faster with agent)

basically, the idea is "do what's fasted by default" (including the docker agent enabled fix :-D), and only disable agent for odd cases.

How about something like this?

```go
optionsMap := map[string]bool{
	"sudo": false,
	"disable-agent": false,
}
for _, o := range strings.Split(options, ",") {
	if _, ok := optionsMap[o] ; !ok {
		return fmt.Errorf("invalid option: %s", o)
	}
	optionsMap[o] = true
}

if optionsMap["sudo"] {
	// enable sudo
}

if hst.Type() != "localhost" && ! options["disable-agent"] {
	// enable agent
}
```

Reply:
That looks like a more elegant solution indeed.

Reply:
:heavy_check_mark: